x/evm/keeper/code_update.go

```diff
@@ -0,0 +1,91 @@
+package keeper
+
+import (
+	"encoding/hex"
+	"fmt"
+	ethcmn "github.com/ethereum/go-ethereum/common"
+	sdk "github.com/okex/exchain/libs/cosmos-sdk/types"
+	"github.com/okex/exchain/x/evm/types"
+)
+
+var (
+	EventTypeContractUpdateByProposal = "contract-update-by-proposal"
+)
+
+// UpdateContractBytecode update contract bytecode
+func (k *Keeper) UpdateContractBytecode(ctx sdk.Context, p types.ManagerContractByteCodeProposal) sdk.Error {
+	contract := ethcmn.BytesToAddress(p.Contract)
+	substituteContract := ethcmn.BytesToAddress(p.SubstituteContract)
+
+	revertContractByteCode := p.Contract.String() == p.SubstituteContract.String()
+
+	preCode := k.EvmStateDb.GetCode(contract)
+	ContractAddr := k.EvmStateDb.GetAccount(contract)
+	if ContractAddr == nil {
+		return types.ErrNotContracAddress(fmt.Errorf("%s", contract.String()))
+	}
+	preCodeHash := ContractAddr.CodeHash
+
+	var newCode []byte
+	if revertContractByteCode {
+		newCode = k.getInitContractCode(ctx, p.Contract)
+		if len(newCode) == 0 {
+			return nil
+		}
+	} else {
+		newCode = k.EvmStateDb.GetCode(substituteContract)
+	}
+	// update code
+	k.EvmStateDb.SetCode(contract, newCode)
```

Comment:
suggest not to use EvmStateDb, this create by mpt_pr. mpt_pr disable forever. I'm not sure EvmStateDb weather exist others data, so use CreateEmptyCommitStateDB